packages/@n8n/task-runner-python/src/task_executor.py

```diff
@@ -107,40 +146,32 @@ def execute_process(
                 assert process.exitcode is not None
                 raise TaskSubprocessFailedError(process.exitcode)
 
-            try:
-                returned = queue.get_nowait()
-            except Empty:
-                raise TaskResultMissingError()
-            except EOFError as e:
-                logger.error(f"Failed to retrieve results from child process: {e}")
+            pipe_reader.join(timeout=PIPE_READER_JOIN_TIMEOUT)
+
+            if pipe_reader.is_alive():
```

Comment:
I'm a little worried on the behavior of the rest of the function here if we meet the timeout and end up racing the write here - we seem to handle things somewhat gracefully even if a lot of data is being written to `pipe_message_list` so I can't spot anything inherently wrong here, but voicing a small concern anyways 🤔

Probably this works just fine in practice, and seems to work well locally.